internal/apex/apex.go

```diff
@@ -59,6 +69,14 @@ type Apex struct {
 	hostname             string
 	symmetricNat         bool
 	logger               *zap.SugaredLogger
+	// See the ApexStatus* constants
+	status    int
+	statusMsg string
+	version   string
+	// Used to tell the CtlServer to stop for shutdown
+	ctlServerStop chan struct{}
+	// Used during shutdown to wait for CtlServer to stop
+	wg sync.WaitGroup
```

Comment:
You don't need ctlServerStop or wg fields here.  You can pass a ctx and wg as arguments to the CtlServerStart function.